api/v1beta1/network_disruption.go

```diff
@@ -358,20 +364,24 @@ func (s *NetworkDisruptionSpec) GenerateArgs() []string {
 
 	// append hosts
 	for _, host := range s.Hosts {
-		if host.DNSResolver == "" {
-			args = append(args, "--hosts", fmt.Sprintf("%s;%d;%s;%s;%s", host.Host, host.Port, host.Protocol, host.Flow, host.ConnState))
-		} else {
-			args = append(args, "--hosts", fmt.Sprintf("%s;%d;%s;%s;%s;%s", host.Host, host.Port, host.Protocol, host.Flow, host.ConnState, host.DNSResolver))
+		hostStr := fmt.Sprintf("%s;%d;%s;%s;%s;%s", host.Host, host.Port, host.Protocol, host.Flow, host.ConnState, host.DNSResolver)
+
+		if host.Percentage != nil {
+			hostStr = fmt.Sprintf("%s;%d", hostStr, *host.Percentage)
 		}
+
+		args = append(args, "--hosts", hostStr)
 	}
 
 	// append allowed hosts
 	for _, host := range s.AllowedHosts {
-		if host.DNSResolver == "" {
-			args = append(args, "--allowed-hosts", fmt.Sprintf("%s;%d;%s;%s;%s", host.Host, host.Port, host.Protocol, host.Flow, host.ConnState))
-		} else {
-			args = append(args, "--allowed-hosts", fmt.Sprintf("%s;%d;%s;%s;%s;%s", host.Host, host.Port, host.Protocol, host.Flow, host.ConnState, host.DNSResolver))
+		hostStr := fmt.Sprintf("%s;%d;%s;%s;%s;%s", host.Host, host.Port, host.Protocol, host.Flow, host.ConnState, host.DNSResolver)
+
+		if host.Percentage != nil {
```

Comment:
same comment as the last pr, why not just add it by default as we did before?

Reply:
I do agree with you. I will do it 🫡

Reply:
Done ✔️